kratos/tests/test_monotonicity_preserving_solver.py

```diff
@@ -49,8 +49,8 @@ def test_MonotonicityPreservingSolver(self):
         KratosMultiphysics.ReadMatrixMarketVector(rhs_file_path,b)
         n = A.Size1()
         x = KratosMultiphysics.Vector(n)
-        for dof in x:
-            dof = 300.0
+        for i in range(0,x.Size()):
```

Comment:
I am not clear of what you do here...you will override it in 55 no?

Reply:
because whatever value im adding on LHS to remove non-diagonal terms, I also add it to RHS as -prod(LHS, dofs) so they have to be initialized.